src/DGmethods/test/Sphere/advection_sphere.jl

```diff
@@ -0,0 +1,290 @@
+# Advection on the Sphere test case.  This is a case similar to the one presented in:
+#
+#@article{WILLIAMSON1992211,
+#title = "A standard test set for numerical approximations to the shallow water equations in spherical geometry",
+#journal = "Journal of Computational Physics",
+#volume = "102",
+#number = "1",
+#pages = "211 - 224",
+#year = "1992",
+#issn = "0021-9991",
+#doi = "https://doi.org/10.1016/S0021-9991(05)80016-6",
+#url = "http://www.sciencedirect.com/science/article/pii/S0021999105800166",
+# }
+#
+# This version runs the advection on the sphere as a stand alone test (no dependence
+# on CLIMA moist thermodynamics)
+
+using MPI
+using CLIMA.Topologies
+using CLIMA.Grids
+using CLIMA.DGBalanceLawDiscretizations
+using CLIMA.DGBalanceLawDiscretizations.NumericalFluxes
+using CLIMA.MPIStateArrays
+using CLIMA.LowStorageRungeKuttaMethod
+using CLIMA.ODESolvers
+using CLIMA.GenericCallbacks
+using LinearAlgebra
+using StaticArrays
+using Logging, Printf, Dates
+
+const uid, vid, wid = 1:3
+const integration_testing = true
+const γ_exact = 7 // 5
+
+if !@isdefined integration_testing
+  const integration_testing =
+    parse(Bool, lowercase(get(ENV,"JULIA_CLIMA_INTEGRATION_TESTING","false")))
+  using Random
+end
+
+# preflux computation: NOT needed for this test
+@inline function preflux(Q, _...)
+  γ::eltype(Q) = γ_exact
+  @inbounds ρ = Q[1]
+    ρinv = 1 / ρ
+    u, v, w, P = 0, 0, 0, 0
+    (P, u, v, w, ρinv)
+end
+
+#{{{ advectionflux
+advectionflux!(F, Q, QV, aux, t) = advectionflux!(F, Q, QV, aux, t, preflux(Q)...)
+@inline function advectionflux!(F,Q,QV,aux,t,P,u,v,w,ρinv)
+    @inbounds begin
```

Comment:
Here and elsewhere: According to the [CLIMA coding convention](https://climate-machine.github.io/CLIMA/latest/CodingConventions/) 2 spaces (not 4) should be used.